Assignments/Splitwise/Python/ProcessInput.py

```diff
@@ -0,0 +1,76 @@
+from user import user
+
+class ProcessInput:
+    def __init__(self):
+        self.users = dict()
+        self.owes = dict()
+
+    # process 'ADD' function
+    def processAdd(self,inputList : List[str]):
+        usersString = inputList[2:]
+        count = 1
+        for usr in usersString:
+            id = 'u'+str(count)
+            self.users[id] = user(id,usr)
+            count += 1
+    
+    #process 'EXPENSE' function
+    def processExpense(self, inputList: List[str]):
+        owedTo = inputList[1]
+        amount = int(inputList[2])
+        num = int(inputList[3])
+        owedBy = inputList[4:4+num]
+
+        # Update the transaction 
+        def updateAmount(owedTo, usr, amt) :
```

Comment:
Why is this nested ? You should consider know ing the implications and disadvantages of nested functions before you start implementing them.

Reply:
I have removed this file from previous approach.